cmd/eks.go

```diff
@@ -230,6 +230,35 @@ If max-retries is unspecified, this command will use a value that translates to
 					waitSleepBetweenRetriesFlag,
 				},
 			},
+			cli.Command{
+				Name:  "drain",
+				Usage: "Drain all Pods from all instances in the provided Auto Scaling Groups.",
+				Description: `Drain Pods from the instances in the provided Auto Scaling Groups. This can be used to gracefully retire existing Auto Scaling Groups by ensuring the Pods are evicted in a manner that respects disruption budgets.
+
+You can read more about the drain operation in the official documentation: https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/.
+
+To drain the Auto Scaling Group "my-asg" in the region "us-east-2":
+
+  kubergrunt eks drain --asg-name my-asg --region us-east-2
+
+You can also drain multiple ASGs by providing the "--asg-name" option multiple times:
+
+  kubergrunt eks drain --asg-name my-asg-a --asg-name my-asg-b --asg-name my-asg-b --region us-east-2
+`,
+				Action: drainASG,
+				Flags: []cli.Flag{
+					clusterRegionFlag,
+					clusterAsgNameFlag,
+					eksKubectlContextNameFlag,
+					genericKubeconfigFlag,
+					genericKubectlServerFlag,
+					genericKubectlCAFlag,
+					genericKubectlTokenFlag,
+					genericKubectlEKSClusterArnFlag,
```

Comment:
Hm, are all these flags here in case you don't already have the kube config pointing to the right context?

Reply:
This is the standard set of flags we provide for authenticating `kubergrunt` commands to `kubectl`. All commands that need to make kubernetes calls need to have all of these flags.